monai/bundle/workflows.py

```diff
@@ -0,0 +1,268 @@
+                settings_ = DEFAULT_EXP_MGMT_SETTINGS[tracking]
+            else:
+                settings_ = ConfigParser.load_config_files(tracking)
+            self.patch_bundle_tracking(parser=self.parser, settings=settings_)
+
+    def initialize(self) -> Any:
+        # reset the "reference_resolver" buffer at initialization stage
+        self.parser.parse(reset=True)
+        return self._run_expr(id=self.init_id)
+
+    def run(self) -> Any:
+        return self._run_expr(id=self.run_id)
+
+    def finalize(self) -> Any:
+        return self._run_expr(id=self.final_id)
+
+    def check_properties(self):
+        super().check_properties()
+        # also check whether the optional properties use correct ID name if existing
+        wrong_props = []
+        for n, p in self.properties.items():
+            if not p[BundleProperty.REQUIRED] and not self._check_optional_id(name=n, property=p):
+                wrong_props.append(n)
+        if wrong_props:
+            raise ValueError(f"loaded bundle defines the following optional properties with wrong ID: {wrong_props}")
+        # check validation `interval` property
+        if "train#handlers" in self.parser:
+            for h in self.parser["train#handlers"]:
+                if h["_target_"] == "ValidationHandler":
+                    interval = h.get("interval", None)
+                    if interval is not None and interval != "val_interval":
+                        raise ValueError("please use ID 'val_interval' to define validation interval in training.")
+
+    def _run_expr(self, id: str, **kwargs) -> Any:
+        return self.parser.get_parsed_content(id, **kwargs) if id in self.parser else None
+
+    def _get_property(self, name: str, property: dict):
+        if not self.parser.ref_resolver.is_resolved():
+            raise RuntimeError("please execute 'initialize' before getting any parsed content.")
+        id = property[BundlePropertyConfig.ID]
+        if id not in self.parser:
+            if not property[BundleProperty.REQUIRED]:
+                return None
+            else:
+                raise KeyError(f"property '{name}' with config ID '{id}' not in the config.")
+        return self.parser.get_parsed_content(id=id)
+
+    def _set_property(self, name: str, property: dict, value: Any):
+        id = property[BundlePropertyConfig.ID]
+        if id not in self.parser:
+            if not property[BundleProperty.REQUIRED]:
+                return None
+            else:
+                raise KeyError(f"property '{name}' with config ID '{id}' not in the config.")
+        self.parser[id] = value
+        # must parse the config again after changing the content
+        self.parser.ref_resolver.reset()
+
+    def _check_optional_id(self, name: str, property: dict):
+        if BundlePropertyConfig.REF_ID not in property:
+            raise ValueError(f"can't find the ID of reference config item for optional property '{name}'.")
```

Comment:
```suggestion
            raise ValueError(f"Cannot find the ID of reference config item for optional property '{name}'.")
```

Reply:
updated in the latest commit.

Thanks.